### DISCUSSION TITLE
Question for sequential activation of DMPs for multiple subtasks

### DISCUSSION THREAD

**Discussion by jqloh** [dfki-ric/movement_primitives#63] [Q&A]
Hi, I would like to ask if there are any references on how we can implement this library for multiple activations for different DMPs for a complex task composed of different subtasks? 

It is recommended to implement a library of DMPs for complex tasks due to its inability to handle different via points.

**Comment by AlexanderFabisch** [comment]
Hi @jqloh ,

Wouldn't a sequence of multiple DMPs work? The only drawback is that transitions between DMPs might not be smooth, since the velocity and acceleration is usually 0 at the end of a DMP. There are DMP variants that support a smooth transition, but they are not implemented here (yet). If it is about multiple viapoints, there is also the experimental [state following DMP](https://github.com/dfki-ric/movement_primitives/blob/main/movement_primitives/dmp/_state_following_dmp.py).
